File: lib/src/main/java/com/otaliastudios/transcoder/internal/transcode/TranscodeDispatcher.java
package com.otaliastudios.transcoder.internal.transcode;

import android.os.Handler;

import androidx.annotation.NonNull;

import com.otaliastudios.transcoder.TranscoderListener;
import com.otaliastudios.transcoder.TranscoderOptions;

/**
 * Wraps a TranscoderListener and posts events on the given handler.
 */
class TranscodeDispatcher {

    private final Handler mHandler;
    private final TranscoderListener mListener;

    TranscodeDispatcher(@NonNull TranscoderOptions options) {
        mHandler = options.getListenerHandler();
        mListener = options.getListener();
    }

    void dispatchCancel() {
        mHandler.post(new Runnable() {
            @Override
            public void run() {
                mListener.onTranscodeCanceled();
            }
        });
    }

    void dispatchSuccess(final int successCode) {
        mHandler.post(new Runnable() {
            @Override
            public void run() {
                mListener.onTranscodeCompleted(successCode);
            }
        });
    }

    void dispatchFailure(@NonNull final Throwable exception) {
        mHandler.post(new Runnable() {
            @Override
            public void run() {
                mListener.onTranscodeFailed(exception);
            }
        });
    }

    void dispatchProgress(final double progress) {
        mHandler.post(new Runnable() {
            @Override
            public void run() {
                mListener.onTranscodeProgress(progress);
            }
        });
    }
}

File: lib/src/main/java/com/otaliastudios/transcoder/internal/transcode/TranscodeDispatcher.kt
package com.otaliastudios.transcoder.internal.transcode

import com.otaliastudios.transcoder.TranscoderOptions
import com.otaliastudios.transcoder.TranscoderListener

/**
 * Wraps a TranscoderListener and posts events on the given handler.
 */
internal class TranscodeDispatcher(options: TranscoderOptions) {
    private val mHandler = options.listenerHandler
    private val mListener = options.listener
    fun dispatchCancel() {
        mHandler.post { mListener?.onTranscodeCanceled() }
    }

    fun dispatchSuccess(successCode: Int) {
        mHandler.post { mListener?.onTranscodeCompleted(successCode) }
    }

    fun dispatchFailure(exception: Throwable) {
        mHandler.post { mListener?.onTranscodeFailed(exception) }
    }

    fun dispatchProgress(progress: Double) {
        mHandler.post { mListener?.onTranscodeProgress(progress) }
    }

}
